optaplanner-wb-screens/optaplanner-wb-guided-rule-editor/optaplanner-wb-guided-rule-editor-client/src/main/java/org/optaplanner/workbench/screens/guidedrule/client/widget/ConstraintMatchInputWidgetBlurHandler.java

```diff
@@ -16,22 +16,94 @@
+                    case "org.optaplanner.core.api.score.buildin.hardmediumsoft.HardMediumSoftScoreHolder": {
+                        try {
+                            Integer.parseInt(inputValue);
+                        } catch (NumberFormatException e) {
+                            widget.showError(translationService.getTranslation(ConstraintMatchInputWidgetBlurHandler_IntegerValueParsingError));
+                            return;
+                        }
+                        break;
+                    }
+                    // long
+                    case "org.optaplanner.core.api.score.buildin.bendablelong.BendableLongScoreHolder":
+                    case "org.optaplanner.core.api.score.buildin.simplelong.SimpleLongScoreHolder":
+                    case "org.optaplanner.core.api.score.buildin.hardsoftlong.HardSoftLongScoreHolder":
+                    case "org.optaplanner.core.api.score.buildin.hardmediumsoftlong.HardMediumSoftLongScoreHolder": {
+                        try {
+                            Long.parseLong(inputValue);
+                        } catch (NumberFormatException e) {
+                            widget.showError(translationService.getTranslation(ConstraintMatchInputWidgetBlurHandler_LongValueParsingError));
+                            return;
+                        }
+                        break;
+                    }
+                    // double
+                    case "org.optaplanner.core.api.score.buildin.simpledouble.SimpleDoubleScoreHolder":
+                    case "org.optaplanner.core.api.score.buildin.hardsoftdouble.HardSoftDoubleScoreHolder": {
+                        try {
+                            Double.parseDouble(inputValue);
+                        } catch (NumberFormatException e) {
+                            widget.showError(translationService.getTranslation(ConstraintMatchInputWidgetBlurHandler_DoubleValueParsingError));
+                            return;
+                        }
+                        break;
+                    }
+                    // java.math.BigDecimal
+                    case "org.optaplanner.core.api.score.buildin.bendablebigdecimal.BendableBigDecimalScoreHolder":
+                    case "org.optaplanner.core.api.score.buildin.simplebigdecimal.SimpleBigDecimalScoreHolder":
+                    case "org.optaplanner.core.api.score.buildin.hardsoftbigdecimal.HardSoftBigDecimalScoreHolder":
+                    case "org.optaplanner.core.api.score.buildin.hardmediumsoftbigdecimal.HardMediumSoftBigDecimalScoreHolder": {
+                        try {
+                            new BigDecimal(inputValue);
+                        } catch (NumberFormatException e) {
+                            widget.showError(translationService.getTranslation(ConstraintMatchInputWidgetBlurHandler_BigDecimalValueParsingError));
+                            return;
+                        }
+                        break;
+                    }
```

Comment:
(merging without waiting for this suggestion)

Ideally, we want a `default` case in this switch that `throws new IllegalStateException("impossible state for scoreHolderType (" + scoreHolderType + ") )`, so it fails fast if future additional scoreHolderTypes aren't introduced properly. That would be an exception that is never thrown (in theory).